### PR TITLE
fix: app actions structure is wrong for export

### PR DESCRIPTION
**Pull request by spaenleh**
In this PR: I fix the structure of the apps file in the exported zip.

Now the structure is as follows:
 - In JSON, there will be a single file `apps.json` that will contain:
    ```json
    {
      "actions": [ array of actions ],
      "data": [ array of data ],
      "settings": [ array of data ],
    }
    ```
- In CSV, there will be 3 files, `app_actions.csv`, `app_data.csv` and `app_settings.csv` which will contain tabular data from all items.

### CODE REVIEW

File: src/services/action/utils/export-utils.test.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+const writeFileSyncMock = jest.spyOn(fs, 'writeFileSync').mockImplementation(() => {});
+describe('writeFileForFormat', () => {
+  describe('CSV format', () => {
+    it('Empty array should not write anything', () => {
+      writeFileForFormat('./test.csv', ExportActionsFormatting.CSV, []);
+      expect(writeFileSyncMock).not.toHaveBeenCalled();
+    });
+
+    it('Object should not write anything', () => {
```
Comment (ReidyT) [inline src/services/action/utils/export-utils.test.ts:15]:
```suggestion
    it('Empty object should not write anything', () => {
```

File: src/services/action/utils/export-utils.test.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    });
+  });
+
+  describe('JSON format', () => {
+    it('Empty array should not write anything', () => {
+      writeFileForFormat('./test.json', ExportActionsFormatting.JSON, []);
+      expect(writeFileSyncMock).not.toHaveBeenCalled();
+    });
+
+    it('Object should not write anything', () => {
```
Comment (ReidyT) [inline src/services/action/utils/export-utils.test.ts:46]:
```suggestion
    it('Empty object should not write anything', () => {
```

File: src/services/action/utils/export.ts [hunk 3]
```diff
@@ -40,14 +43,15 @@ const flattenObject = (obj: RecursiveObject, prefix: string = ''): ReturnObject
   }, {});
 };
 
-const writeFileForFormat = (
+export const writeFileForFormat = (
   path: string,
   format: ExportActionsFormatting,
-  data: object[], // TODO: Replace 'object' with specific type(s) when defining schema for data to export
+  // TODO: Replace 'object' with specific type(s) when defining schema for data to export
```
Comment (ReidyT) [inline src/services/action/utils/export.ts:49]: Should we create an issue?

File: src/services/action/utils/export.ts [hunk 4]
```diff
@@ -130,24 +141,58 @@ export const exportActionsInArchive = async (args: {
...
+      case ExportActionsFormatting.JSON: {
+        // create files for the apps
+        const appsPath = path.join(
+          fileFolderPath,
+          buildActionFileName('apps', archiveDate, format),
+        );
+        writeFileForFormat(appsPath, format, { appActions, appData, appSettings });
+        break;
+      }
+      // For CSV format there will be one file for actions, one fofr data and one for settings
```
Comment (ReidyT) [inline src/services/action/utils/export.ts:170]:
```suggestion
      // For CSV format there will be one file for actions, one for data and one for settings
```

File: src/services/action/utils/export.ts [hunk 5]
```diff
@@ -130,24 +141,58 @@ export const exportActionsInArchive = async (args: {
...
+          fileFolderPath,
+          buildActionFileName('app_data', archiveDate, format),
+        );
+        writeFileForFormat(appDataPath, format, appData);
+
+        const appSettingsPath = path.join(
+          fileFolderPath,
+          buildActionFileName('app_settings', archiveDate, format),
+        );
+        writeFileForFormat(appSettingsPath, format, appSettings);
```
Comment (ReidyT) [inline src/services/action/utils/export.ts:190]: Should we refactor in another function to then call it for the three CSV exports, as well as for the JSON export?

File: src/services/item/plugins/action/service.ts [hunk 6]
```diff
@@ -207,12 +207,12 @@ export class ActionItemService {
...
     } = {};
-    for (const { id: appId } of [item, ...descendants].filter(
-      ({ type }) => type === ItemType.APP,
-    )) {
+    const appItems = [item, ...descendants].filter(({ type }) => type === ItemType.APP);
+    for (const { id: appId } of appItems) {
       const appData = await repositories.appDataRepository.getForItem(
         appId,
         {},
+        // needs investigating: does this mean a reader could export the actions of an item and see all users responses ?
```
Comment (ReidyT) [inline src/services/item/plugins/action/service.ts:215]: Only admin should be allowed to do that right?

**Review by ReidyT — APPROVED** [review]
LGTM

File: src/services/action/utils/export.ts (hunk 5 above)
Comment (spaenleh) [reply src/services/action/utils/export.ts:190]: I don't know, is the function too lengthy like this ? Not sure refactoring to a different function will make it more readable.

File: src/services/item/plugins/action/service.ts (hunk 6 above)
Comment (spaenleh) [reply src/services/item/plugins/action/service.ts:215]: Add an issue for this.

Comment (swouf) [reply src/services/item/plugins/action/service.ts:215]: Yes, we cannot allow a reader to get access to other users' data with the export. We may give them the possibility to export their own data.

File: src/services/action/utils/export.ts (hunk 5 above)
Comment (ReidyT) [reply src/services/action/utils/export.ts:190]: I'm okay with what your did too ^^

**Review by swouf — APPROVED** [review]

File: src/services/item/plugins/action/schemas.ts [hunk 7]
```diff
@@ -128,7 +130,7 @@ export const getAggregateActions = {
   },
 };
 
-export const exportAction = {
+export const exportAction: FastifySchema = {
```
Comment (Defferrard) [inline src/services/item/plugins/action/schemas.ts:133]: If we want to be able to infer type in the handler we should not type the schema to FastifySchema. You should do `as const satisfies FastifySchema`

This schema will be refactored anyway

**Review by Defferrard — APPROVED** [review]

**Pull request merged**